lib/main.dart

```diff
@@ -289,35 +298,31 @@ class _TenTenOneState extends State<TenTenOneApp> {
       final appSupportDir = await getApplicationSupportDirectory();
       FLog.info(text: "App data will be stored in: " + appSupportDir.toString());
 
-      await api.initWallet(path: appSupportDir.path);
-      await api.initDb(appDir: appSupportDir.path);
-
       final isUserSeedBackupConfirmed =
           await TenTenOneSharedPreferences.instance.isUserSeedBackupConfirmed();
       final seedBackupModel = context.read<SeedBackupModel>();
       seedBackupModel.update(isUserSeedBackupConfirmed);
 
+      final walletChangeNotifier = context.read<WalletInfoChangeNotifier>();
+      final tentenoneChangeNotifier = context.read<TenTenOneChangeNotifier>();
+
       FLog.info(text: "Starting ldk node");
-      api
-          .runLdk()
-          .then((value) => FLog.info(text: "ldk node stopped."))
-          .catchError((error) => FLog.error(text: "ldk stopped with an error", exception: error));
-
-      setState(() {
-        FLog.info(text: "TenTenOne is ready!");
-        ready = true;
+      api.run(appDir: appSupportDir.path).listen((event) {
+        if (event is Event_Ready) {
+          tentenoneChangeNotifier.ready();
+        } else if (event is Event_Offer) {
+          cfdOffersChangeNotifier.update(event.field0);
+        } else if (event is Event_WalletInfo) {
+          walletChangeNotifier.update(event.field0);
+        } else if (event is Event_Init) {
+          tentenoneChangeNotifier.set(event.field0);
+        }
       });
```

Comment:
A switch statement would be better, with logging an error for unhandled event. (or crashing the whole thing :P I for one would be for explicit handling of all of these.)

Reply:
I tried with a switch, but did not find a way of switching over an instance of. Do you have a suggestion?

Reply:
aaaaaa. well,  in that case, mabye we should just log an error on `else`? especially if we can print the instance of an unrecognised event?

Reply:
sounds good!